### PR TITLE
Implement REST API

### PR DESCRIPTION
**Pull request by RonAzar**
Added GET, POST, PUT, and DELETE endpoints.
Tested API using Postman.

### CODE REVIEW

**Review by brainpicker — COMMENTED** [review]
Some minor comments. The code is very organized. I liked that you went above and beyond to handle id's and faking the list yourself.

File: src/main/kotlin/com/hibob/academy/rest_api/PetController.kt [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        val pet = pets.find { it.petId == petId }
+        return when {
+            pet == null -> {
+                throw NotFoundException("Pet not found")  // Pet not found
+            }
+            petId == 123L -> {
+                Response.status(Response.Status.UNAUTHORIZED).build()  // Unauthorized for petId 123
+            }
+            else -> {
+                Response.ok("Pet Type: ${pet.type}, Name: ${pet.name}, Company ID: ${pet.companyId}, Date of Arrival: ${pet.dateOfArrival}").build()
```
[the remark this answers is not in the document]
Comment (RonAzar) [reply src/main/kotlin/com/hibob/academy/rest_api/PetController.kt:34]: You’re right. I’ll fix it now.

File: src/main/kotlin/com/hibob/academy/rest_api/OwnerResource.kt [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+val owners = mutableListOf(
+    Owner(1L,"Ron", 101L, 10L),
+    Owner(2L,"Or", 102L, 11L),
+    Owner(3L,"Noam", 103L, 12L)
+)
+
+
+@Controller
+@Path("/api/ron/owners/envelopes")
```
[the remark this answers is not in the document]
Comment (RonAzar) [reply src/main/kotlin/com/hibob/academy/rest_api/OwnerResource.kt:16]: Gady, I’m not sure which line contains the mistake. Could you please clarify?

File: src/main/kotlin/com/hibob/academy/rest_api/OwnerResource.kt [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    fun getAll() : Response {
+        return Response.ok(owners).build()
+    }
+
+    //GET: Retrieve owner by ownerId
+    @GET
+    @Path("/{ownerId}")
+    fun getById(@PathParam("ownerId") ownerId: Long): Response {
+        val owner = owners.find { it.ownerId == ownerId }
+        return when{
```
[the remark this answers is not in the document]
Comment (RonAzar) [reply src/main/kotlin/com/hibob/academy/rest_api/OwnerResource.kt:31]: Thanks for the comment, Gady! The Unauthotized handling is the same as what I did in the pet methods—nothing special was checked.

File: src/main/kotlin/com/hibob/academy/rest_api/OwnerResource.kt [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+            throw NotFoundException("No owner with id $ownerId")
+        }
+    }
+
+    //DELETE: Delete an owner by ownerId
+    @DELETE
+    @Path("/{ownerId}")
+    fun deleteOwner(@PathParam("ownerId") ownerId: Long): Response {
+        val owner = owners.find { it.ownerId == ownerId }
+        return if(owner != null) {
```
[the remark this answers is not in the document]
Comment (RonAzar) [reply src/main/kotlin/com/hibob/academy/rest_api/OwnerResource.kt:69]: You’re right. I’m fixing it right now.

File: src/main/kotlin/com/hibob/academy/rest_api/OwnerResource.kt [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    //DELETE: Delete an owner by ownerId
+    @DELETE
+    @Path("/{ownerId}")
+    fun deleteOwner(@PathParam("ownerId") ownerId: Long): Response {
+        val owner = owners.find { it.ownerId == ownerId }
+        return  owner?.let {
+            owners.remove(owner)
+            Response.ok("${owner.toString()} Has been removed").build()
+        }?: throw NotFoundException("No owner with id $ownerId")
+//        return if(owner != null) {
```
Comment (brainpicker) [inline src/main/kotlin/com/hibob/academy/rest_api/OwnerResource.kt:73]: You can delete this comment now :)

**Review by brainpicker — APPROVED** [review]
Nice, Just please watch the styling. There is a lot of mess and inconsistency. The IntelliJ allows to re-format before committing in the dialog. Or you can just format the code yourself manually.

File: src/main/kotlin/com/hibob/academy/rest_api/PetController.kt [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+import java.util.Date
+
+val pets = mutableListOf(
+    Pet(1L, "Buddy", "Dog", 101, Date()),
+    Pet(2L, "Whiskers", "Cat", 102, Date()),
+    Pet(3L, "Polly", "Parrot", 103, Date())
+)
+
+@Controller
+@Path("/api/ron/pets/envelopes")
```
Comment (brainpicker) [inline src/main/kotlin/com/hibob/academy/rest_api/PetController.kt:16]: No need envelopes here either.

**Review by brainpicker — APPROVED** [review]

File: src/main/kotlin/com/hibob/academy/rest_api/PetController.kt (hunk 6 above)
Comment (RonAzar) [reply src/main/kotlin/com/hibob/academy/rest_api/PetController.kt:16]: Fixed!

**Pull request merged**